src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1015,7 +1023,23 @@ static int rbd_nbd(int argc, const char *argv[])
   int r;
   Config cfg;
   vector<const char*> args;
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+
   argv_to_vec(argc, argv, args);
+  env_to_vec(args);
+
+  for (auto &temp_args : args) {
+	if (strcmp(temp_args, "map") == 0) {
+      code_env = CODE_ENVIRONMENT_DAEMON;
+      env_flag = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+	  break;
+	}
+  }
+
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, code_env, env_flag);
+  g_ceph_context->_conf->set_val_or_die("pid_file", "");
```

Comment:
@liupan1111 I am not very happy you returned `global_init` back to `rbd_nbd`, that is run for all commands. We had some reasons to call it for "map" command only: it made the code cleaner avoiding e.g. manual command line parsing as you added above to check if it is "map" command.

Actually, the problem is that `parse_args`, which has to filter out all ceph common options before parsing nbd specific options, does not filter out "-c" like options. These can be filtered out by `ceph_argparse_early_args`. So if you just add something like below at the beginning of `parse_args`:
```
std::string conf_file_list;
std::string cluster;
CephInitParameters iparams = ceph_argparse_early_args(
    args, CEPH_ENTITY_TYPE_CLIENT, &cluster, &conf_file_list);
```
I think it should fix the issue.

The only potential problem with not calling `global_init` for non "map" commands is that if someone specifies "-c" like option for such a command it will be just ignored. This may have undesirable effect e.g. if the user has different default pool specified in config, and then "list" command will output mapped with wrong default pool. Still I prefer this solution.

Reply:
yes! This method is much better. Thanks.